git/cmd.py

```diff
@@ -902,8 +902,12 @@ def transform_kwarg(self, name, value, split_single_char_options):
 
     def transform_kwargs(self, split_single_char_options=True, **kwargs):
         """Transforms Python style kwargs into git command line options."""
+        # only since 3.6 Python preserves the order of kwargs and thus has a stable
+        # order. For older versions sort the kwargs by the key to get a stable
+        # order.
+        if sys.version_info[:2] < (3, 6):
```

Comment:
In order to make people aware, could you add a deprecation warning mentioning the date until this will work?